File: cpp/0263-ugly-number.cpp
class Solution {
public:
    bool isUgly(int n) {
        if(n <= 0)
            return false;
        
        for(int p: {2, 3, 5})
            while(n % p == 0)
                n = n / p;
        return n == 1;
    }
};

File: swift/0263-ugly-number.swift
class Solution {
    func isUgly(_ n: Int) -> Bool {
        if n <= 0 {
            return false
        }

        var num = n
        for p in [2, 3, 5] {
            while num % p == 0 {
                num /= p
            }
        }
        return num == 1
    }
}